### ISSUE TITLE
EVM/smart contract pallet

### ISSUE THREAD

**Issue by mochet** [Joystream/joystream#5169]
# Problem
Described here: https://pioneerapp.xyz/#/forum/thread/924

# Solution
Add EVM/smart contract pallet to Joystream

**Comment by kdembler** [comment]
I like the idea but just to get on the same page, this is hard to do. Back in the days we've considered using EVM for some core functionalities as content directory but from my understanding the conclusion was that it was too complex for the benefits it brings. One of the related issues here: https://github.com/Joystream/joystream/issues/1816

**Comment by bedeho** [comment]
The issue is the integration between EVM and native pallets, but we did not really push this hard. There are many live production examples of how to do this integratino, like Astar, Acala, Moonbeam, etc. However, what you have to ask yourself is the _value_, I dont think this is a good use of resources now.

**Comment by mochet** [comment]
> However, what you have to ask yourself is the _value_, I dont think this is a good use of resources now.

Agreed - my intention was for this to possibly be a roadmap item for the distant future, not really right now. But the comments here are helpful and do make me understand that this is definitely not a small endeavor.
